models/airdrop.go

```diff
@@ -23,6 +23,48 @@ type AirdropClaim struct {
 	Claimed       bool    `json:"claimed" gorm:"not null" sql:"DEFAULT:0"`
 }
 
+func GetAirdrop(airdrop *Airdrop) Airdrop {
+	return Airdrop{
+		ID:           airdrop.ID,
+		TokenID:      airdrop.TokenID,
+		MerkleRoot:   airdrop.MerkleRoot,
+		RawData:      airdrop.RawData,
+		OnChainIndex: airdrop.OnChainIndex,
+	}
+}
+
+func GetAirdropClaim(airdropClaim *AirdropClaim) AirdropClaim {
+	return AirdropClaim{
+		WalletAddress: airdropClaim.WalletAddress,
+		ClaimAmount:   airdropClaim.ClaimAmount,
+		AirdropID:     airdropClaim.AirdropID,
+		Airdrop:       airdropClaim.Airdrop,
+		Claimed:       airdropClaim.Claimed,
+	}
+}
+
+func GetAirdropByTokenId(tokenId uint64, db gorm.DB) (Airdrop, error) {
+	var a Airdrop
+	if err := db.Where("token_id=?", tokenId).First(&a).Error; err != nil {
```

Comment:
We will have to be careful here because you need to make sure that the airdrop is the latest active airdrop. This might not return the desired airdrop.